geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/SpatialEncoders.scala

```diff
@@ -0,0 +1,25 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import com.vividsolutions.jts.geom._
+
+trait SpatialEncoders {
+  implicit def jtsGeometryEncoder = ExpressionEncoder[Geometry]()
```

Comment:
I'd consider adding these automatically into implicit scope via a mix-in to the `org.locationtech.geomesa.spark` package object. e.g.: 
https://github.com/s22s/raster-frames/blob/54b8b74f6b1971ed0bbbe3ba5696c932d8f8d3fa/src/main/scala/astraea/spark/rasterframes/package.scala#L42

Reply:
IMO it's good to limit implicits in the default scope, because it can be confusing when reading through the code and also slow down compiling time. @metasim is it just a convenience to have them in the default scope, or is there another reason to move them?

Reply:
It actually looks like these aren't currently used anywhere except test code.

Reply:
@elahrvivaz It's an end-user-experience thing that's consistent with what we do in GeoTrellis and RasterFrames. They are only used as implicit parameters to the various spark `Dataset[T]` creation methods, and are akin to the Scala `CanBuildFrom` type classes, ones that the average user shouldn't need to know about. If the user has imported `org.locationtech.geomesa.spark._` they have to have these if they want to used the typed `Dataset[T]` APIs.

Reply:
PS: To be clear, I think they should stay in the `SpatialEncoders` trait, but have that trait mixed into the library's root package object.